ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt

```diff
@@ -85,30 +77,31 @@ private fun Template2MainContent(state: PaywallViewState.Template2, viewModel: P
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.spacedBy(UIConstant.defaultButtonVerticalSpacing, Alignment.CenterVertically),
     ) {
-        IconImage(paywallData = state.paywallData)
-        val localizedConfig = state.paywallData.localizedConfig()
+        state.templateConfiguration.images.iconUri?.let { IconImage(it) }
+        val localizedConfig = state.templateConfiguration.packageConfiguration.defaultInfo.localization
+        val colors = state.templateConfiguration.getColors(isSystemInDarkTheme())
```

Comment:
By passing `isSystemInDarkTheme` wherever we need the colors, this would get recompositioned from this point, instead of recompositioning everything using the view model.

Reply:
Can we instead have an extension so that we can call this with `state.templateConfiguration.getCurrentColors()`?